app/src/main/java/com/realityexpander/tasky/core/presentation/notifications/RemindAtAlarmManager.kt

```diff
@@ -0,0 +1,166 @@
+package com.realityexpander.tasky.core.presentation.notifications
+
+import android.app.AlarmManager
+import android.app.PendingIntent
+import android.content.Context
+import android.content.Intent
+import android.os.Parcelable
+import com.realityexpander.tasky.MainActivity
+import com.realityexpander.tasky.agenda_feature.domain.AgendaItem
+import com.realityexpander.tasky.agenda_feature.domain.IRemindAtAlarmManager
+import com.realityexpander.tasky.core.presentation.broadcastReceivers.AlarmBroadcastReceiver
+import com.realityexpander.tasky.core.presentation.notifications.RemindAtNotificationManager.ALARM_NOTIFICATION_INTENT_ACTION_ALARM_TRIGGER
+import com.realityexpander.tasky.core.presentation.notifications.RemindAtNotificationManager.ALARM_NOTIFICATION_INTENT_EXTRA_AGENDA_ITEM
+import com.realityexpander.tasky.core.presentation.notifications.RemindAtNotificationManager.ALARM_NOTIFICATION_INTENT_EXTRA_ALARM_ID
+import com.realityexpander.tasky.core.util.toUtcMillis
+import com.realityexpander.tasky.core.util.toZonedDateTime
+import logcat.logcat
+import java.time.ZonedDateTime
+import java.time.temporal.ChronoUnit
+
+object RemindAtAlarmManager : IRemindAtAlarmManager {
```

Comment:
This should be a class taking the app context as a parameter. You can then provide it with Hilt